plugins/genesys/fu-genesys-usbhub-firmware.c

```diff
@@ -240,6 +258,9 @@ fu_genesys_usbhub_firmware_parse(FuFirmware *firmware,
 	case ISP_MODEL_HUB_GL3523:
 		static_ts_offset = GENESYS_USBHUB_STATIC_TOOL_STRING_OFFSET_GL3523;
 		break;
+	case ISP_MODEL_HUB_GL3523PLUS:
```

Comment:
Could you want to push an emulation for this hardware? The genesys plugin is woefully undertested right now -- e.g. https://app.codecov.io/gh/fwupd/fwupd/tree/main/plugins%2Fgenesys 

Generating an emulation is pretty easy now: https://github.com/fwupd/fwupd/blob/main/docs/device-emulation.md

Reply:
OK, let me take a try.

1. Upload chip's test.cab to LVFS and move to `Embargo` stage.
2. Tell the `fwupdmgr` to record a device.
3. Install test.cab through `fwupdmgr` and save the record.
4. Test recorded data through `fwupdmgr`.
5. Upload the recorded data to LVFS.

Does any step go wrong?

Reply:
That sounds about right. if you use gnome-firmware you can even click rather than type.

Reply:
May I tag USB hub and HID together? Because some hubs use the HID to update. I can record the .zip but cannot emulation-load well.

* [hub and successed](https://fwupd.org/downloads/38954ace3092f0dff12163e6daf0940dc67ddd3b27d6cd98c8fca6ed4e4d5c7e-TestEVB_GL3525_M.zip/internal)
* [hub+hid and failed](https://fwupd.org/downloads/da59f16b4b7fe32b794e3d3e72be35f3926a3f55c5134e28bfef0bc08431597d-TestEVB_GL3525_C.zip/internal)

Reply:
@adamgene i think the C failure is because you're doing the `FuGenesysHubhidDevice *hid_channel` thing -- I think if you use the "proxy" (e.g. `fu_device_{g|s}et_proxy()`) then it would work. The proxy knows how to collect and provide emulation events and I think switching to use this would fix this bug too.

Reply:
I'm not sure why we do this before. The USB hub uses a physical HID to send report. So, I should `fu_device_set_proxy(hub, hid)` or reverse? Maybe we can fix this in a new PR.

Reply:
I'm guessing you're using `fu_device_add_private_flag(FU_DEVICE(self), FU_DEVICE_PRIVATE_FLAG_REFCOUNTED_PROXY);` too?